hello-world/src/main/java/com/example/hello_world/MainActivity.java

```diff
@@ -0,0 +1,197 @@
+
+    public interface Callbacks {
+        void onSuccess();
+        void onFailure(Exception e);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        okHttpClient = new OkHttpClient.Builder()
+            .connectTimeout(20, TimeUnit.SECONDS)
+            .readTimeout(20, TimeUnit.SECONDS)
+            .build();
+
+        // Kin client is the accounts manager for Kin accounts
+        kinClient = new KinClient(this, Environment.TEST, STUB_APP_ID, HELLO_WORLD);
+
+        // Kin account the is the entity that holds Kins
+        kinAccount = getKinAccount(APP_INDEX);
+
+        // listener for balance changes
+        addBalanceListeners(kinAccount);
+
+        // async function to get kin balance
+        getKinBalance(kinAccount);
+
+        // create the wallet with the Kin account keypair
+        // as it is an async request, at the callback we will be able to transfer Kins and check the account balance
+        onBoardAccount(kinAccount, new Callbacks() {
+            @Override
+            public void onSuccess() {
+                Log.e(TAG, "Onboarding succeeded");
+                transferKin(kinAccount, TARGET_WALLET, AMOUNT_KIN);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                Log.e(TAG, "Onboarding failed");
+            }
+        });
+    }
+
+    public KinAccount getKinAccount(int index) {
+        // the index that is used to get a specific account from the client manager
+        KinAccount kinAccount = kinClient.getAccount(index);
+        try {
+            // create local keypair
+            if (kinAccount == null) {
+                kinAccount = kinClient.addAccount();
+                Log.e(TAG, "Created new account succeeded");
+            }
+        } catch (CreateAccountException e) {
+            e.printStackTrace();
+        }
+
+        return kinAccount;
+    }
+
+    private void onBoardAccount(@NonNull KinAccount account, @NonNull Callbacks callbacks) {
+        // creating the kin account for a specific user
+        if (account != null) {
+            Request request = new Request.Builder()
+                .url(String.format(URL_CREATE_ACCOUNT, account.getPublicAddress()))
+                .get()
+                .build();
+            okHttpClient.newCall(request)
+                .enqueue(new Callback() {
+                    @Override
+                    public void onFailure(@NonNull Call call, @NonNull IOException e) {
+                        fireOnFailure(callbacks, e);
+                    }
+
+                    @Override
+                    public void onResponse(@NonNull Call call, @NonNull Response response) {
+                        int code = response.code();
+                        response.close();
+                        if (code != 200) {
+                            fireOnFailure(callbacks,
```

Comment:
why do you need to fire the callbacks back to the main method?
you can print the success failure here for simplicity.